src/lang/db/swapper.rs

```diff
@@ -57,17 +57,42 @@ impl Display for SwapReason {
 /// It is expected that diagnostics will be refreshed on it as quickly as possible, otherwise
 /// the entire workspace would be recompiled at an undetermined time leading to bad UX delays.
 pub struct AnalysisDatabaseSwapper {
-    last_replace_time: SystemTime,
+    stopwatch: Stopwatch,
     mutations_since_last_replace: u64,
     db_replace_min_interval: Duration,
     db_replace_min_mutations: u64,
 }
 
+#[derive(Default)]
+struct Stopwatch {
+    start_time: Option<SystemTime>,
+    total_elapsed_time: Duration,
+}
+
+impl Stopwatch {
+    fn start(&mut self) {
+        self.start_time = Some(SystemTime::now());
+    }
+
+    fn stop(&mut self) {
+        let Some(start_time) = self.start_time else {
+            return;
+        };
+
+        let elapsed_time = SystemTime::now().duration_since(start_time).unwrap_or_default();
+        self.total_elapsed_time += elapsed_time;
+    }
```

Comment:
We should set `start_time` to `None` and `assert!` at the beginning to avoid incorrect summing up numbers + actually avoiding developer errors